### ISSUE TITLE
Incorrect refactor of "Use uncurry" 

### ISSUE THREAD

**Issue by waddlaw** [ndmitchell/hlint#1062]
```hs
foo :: Int -> Int
foo = bar (\(f, x) -> baz () () . f $ x)
  where
    bar = undefined
    baz = undefined
```

When I run hlint 3.1.5, I get the following message:

```
Suggestion: Use uncurry
Found:
  \ (f, x) -> baz () () . f $ x
Perhaps:
  uncurry ((.) baz () ())
Note: increases laziness
```

However, if I replace it as suggested, it will result in a compile error due to the lack of parentheses. The correct suggestion is `uncurry ((.) (baz () ()))`.

**Comment by ndmitchell** [comment]
Thanks for the report! The code to match was seeing through `.` but not transforming it appropriately. Now fixed and I'll make a fresh release.
